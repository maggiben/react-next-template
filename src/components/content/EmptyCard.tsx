import styled from "styled-components";
import { Heading } from '@fravega-it/bumeran-ds-fvg'

const Card = styled.div`
  display: block;
  width: 100%;
  border-width: 1px;
  border-style: solid;
  border-color: ${({ theme }) => theme.colors.neutral[300]};
  border-radius: ${({ theme }) => theme.borderRadius.m};
`;

const EmptyCard = () => {
  return (
    <Card>
        <Heading>Bienvenido al buscador de clientes</Heading>
    </Card>
  );
};

  export default EmptyCard;
