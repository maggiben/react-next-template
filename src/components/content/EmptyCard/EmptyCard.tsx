import styled from "styled-components";
import { useTranslation } from 'react-i18next';
import { Heading, TextBody } from '@fravega-it/bumeran-ds-fvg'

const Card = styled.div`
  display: block;
  width: 100%;
  border-width: 1px;
  // border-style: solid;
  // border-color: ${({ theme }) => theme.colors.neutral[300]};
  // border-radius: ${({ theme }) => theme.borderRadius.m};
`;

const EmptyCard = () => {
  const { t } = useTranslation();
  return (
    <Card>

    </Card>
  );
};

  export default EmptyCard;
