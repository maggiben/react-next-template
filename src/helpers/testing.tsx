import { default as baseRenderer } from 'react-test-renderer';
import { render, RenderResult } from '@testing-library/react';
import i18n from '@utils/i18n';
import { I18nextProvider } from 'react-i18next';
import { RecoilRoot } from 'recoil';
import { ThemeProvider } from 'styled-components';
import { theme, IDefaultTheme } from '@fravega-it/bumeran-ds-fvg';
import React, { ReactElement } from 'react';
import '@testing-library/jest-dom'

/**
 * Helper used to render components for testing.
 * The function appends the theme to the children.
 * */
interface IRenderer {
  create: (children: ReactElement) => RenderResult;
  baseRenderer: (children: ReactElement) => baseRenderer.ReactTestRenderer;
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme) => RenderResult;
  renderWithI18n: (children: ReactElement) => RenderResult;
}

export const renderer: IRenderer = {
  create: (children: ReactElement): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  baseRenderer: (children: ReactElement): baseRenderer.ReactTestRenderer =>
    baseRenderer.create(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithI18n: (children: ReactElement): RenderResult => 
    render(
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
      </ThemeProvider>
  ),
};