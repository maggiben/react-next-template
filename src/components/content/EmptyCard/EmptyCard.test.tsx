import React from 'react';
import { renderer } from '@helpers/testing';
import '@testing-library/jest-dom';
import EmptyCard from './EmptyCard';
import { expect } from '@jest/globals';

describe('EmptyCard', () => {
  it('renders the EmptyCard component', () => {
    const tree = renderer.create(<EmptyCard />);
    expect(tree).toMatchSnapshot();
  });
});
