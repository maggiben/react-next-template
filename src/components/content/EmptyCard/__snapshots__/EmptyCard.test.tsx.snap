// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyCard renders the EmptyCard component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="EmptyCard__Container-sc-1xwmzog-1 ilsrC"
      >
        <div
          class="EmptyCard__Center-sc-1xwmzog-0 gdueiD"
          data-testid="empty-container"
        >
          <div
            class="EmptyCard__Circle-sc-1xwmzog-2 fBLbGE"
            data-testid="circle"
          >
            <svg
              height="64"
              viewBox="0 0 64 64"
              width="64"
            >
              <path
                clip-rule="evenodd"
                d="M44.404 32.239c0-6.912-5.61-12.522-12.522-12.522s-12.523 5.61-12.523 12.522S24.97 44.76 31.883 44.76c6.912 0 12.522-5.61 12.522-12.522Zm-21.754.331c0-5.278 4.284-9.563 9.563-9.563 5.279 0 9.563 4.285 9.563 9.563 0 5.279-4.284 9.563-9.563 9.563-5.279 0-9.563-4.284-9.563-9.563Z"
                fill="#DABFFF"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M32 19.22c7.29 0 13.185 5.894 13.185 13.185 0 2.722-.829 5.255-2.249 7.361l7.859 7.86a2.543 2.543 0 1 1-3.598 3.598l-7.859-7.86a13.125 13.125 0 0 1-7.362 2.25 13.173 13.173 0 0 1-13.185-13.186c0-7.29 5.895-13.185 13.185-13.185L32 19.22Zm6.391 23.174A11.825 11.825 0 0 1 32 44.264a11.85 11.85 0 0 1-11.86-11.86A11.85 11.85 0 0 1 32 20.546a11.85 11.85 0 0 1 11.86 11.86c0 2.367-.687 4.544-1.87 6.39a11.968 11.968 0 0 1-3.575 3.575l-.024.024Z"
                fill="#642DB3"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M42.179 32.405c0-5.634-4.569-10.179-10.18-10.179-5.61 0-10.178 4.569-10.178 10.179S26.39 42.583 32 42.583s10.179-4.568 10.179-10.178Zm-19.056 0c0-4.9 3.977-8.877 8.877-8.877s8.877 3.977 8.877 8.877S36.9 41.28 32 41.28s-8.877-3.977-8.877-8.876Z"
                fill="#642DB3"
                fill-rule="evenodd"
              />
              <path
                d="M37.941 12.142h-.165l.331-.97a.378.378 0 0 0-.426-.498l-3.409.687c-.284.047-.402.402-.213.615l2.296 2.604c.19.213.545.142.64-.118l.331-.994s.071.094.118.094c11.789 4.072 15.623 15.031 13.232 23.293a.768.768 0 0 0 .521.947c.071 0 .142.023.213.023.332 0 .64-.213.734-.568 2.58-8.947-1.538-20.807-14.226-25.162l.023.047ZM26.058 51.862h.166l-.331.97a.378.378 0 0 0 .426.498l3.409-.687c.284-.047.402-.402.213-.615l-2.296-2.604a.382.382 0 0 0-.64.118l-.331.995s-.071-.095-.118-.095C14.767 46.371 10.933 35.411 13.323 27.15a.768.768 0 0 0-.52-.947c-.071 0-.142-.024-.213-.024a.754.754 0 0 0-.734.568c-2.58 8.948 1.538 20.807 14.226 25.163l-.024-.048Z"
                fill="#642DB3"
              />
            </svg>
          </div>
          <div
            class="Spacing__SpaceTop-sc-11df7di-1 fckOgK"
          />
          <div
            aria-level="3"
            class="sc-kAyceB kJJjGB"
            role="heading"
          >
            No encontramos resultados
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="EmptyCard__Container-sc-1xwmzog-1 ilsrC"
    >
      <div
        class="EmptyCard__Center-sc-1xwmzog-0 gdueiD"
        data-testid="empty-container"
      >
        <div
          class="EmptyCard__Circle-sc-1xwmzog-2 fBLbGE"
          data-testid="circle"
        >
          <svg
            height="64"
            viewBox="0 0 64 64"
            width="64"
          >
            <path
              clip-rule="evenodd"
              d="M44.404 32.239c0-6.912-5.61-12.522-12.522-12.522s-12.523 5.61-12.523 12.522S24.97 44.76 31.883 44.76c6.912 0 12.522-5.61 12.522-12.522Zm-21.754.331c0-5.278 4.284-9.563 9.563-9.563 5.279 0 9.563 4.285 9.563 9.563 0 5.279-4.284 9.563-9.563 9.563-5.279 0-9.563-4.284-9.563-9.563Z"
              fill="#DABFFF"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M32 19.22c7.29 0 13.185 5.894 13.185 13.185 0 2.722-.829 5.255-2.249 7.361l7.859 7.86a2.543 2.543 0 1 1-3.598 3.598l-7.859-7.86a13.125 13.125 0 0 1-7.362 2.25 13.173 13.173 0 0 1-13.185-13.186c0-7.29 5.895-13.185 13.185-13.185L32 19.22Zm6.391 23.174A11.825 11.825 0 0 1 32 44.264a11.85 11.85 0 0 1-11.86-11.86A11.85 11.85 0 0 1 32 20.546a11.85 11.85 0 0 1 11.86 11.86c0 2.367-.687 4.544-1.87 6.39a11.968 11.968 0 0 1-3.575 3.575l-.024.024Z"
              fill="#642DB3"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M42.179 32.405c0-5.634-4.569-10.179-10.18-10.179-5.61 0-10.178 4.569-10.178 10.179S26.39 42.583 32 42.583s10.179-4.568 10.179-10.178Zm-19.056 0c0-4.9 3.977-8.877 8.877-8.877s8.877 3.977 8.877 8.877S36.9 41.28 32 41.28s-8.877-3.977-8.877-8.876Z"
              fill="#642DB3"
              fill-rule="evenodd"
            />
            <path
              d="M37.941 12.142h-.165l.331-.97a.378.378 0 0 0-.426-.498l-3.409.687c-.284.047-.402.402-.213.615l2.296 2.604c.19.213.545.142.64-.118l.331-.994s.071.094.118.094c11.789 4.072 15.623 15.031 13.232 23.293a.768.768 0 0 0 .521.947c.071 0 .142.023.213.023.332 0 .64-.213.734-.568 2.58-8.947-1.538-20.807-14.226-25.162l.023.047ZM26.058 51.862h.166l-.331.97a.378.378 0 0 0 .426.498l3.409-.687c.284-.047.402-.402.213-.615l-2.296-2.604a.382.382 0 0 0-.64.118l-.331.995s-.071-.095-.118-.095C14.767 46.371 10.933 35.411 13.323 27.15a.768.768 0 0 0-.52-.947c-.071 0-.142-.024-.213-.024a.754.754 0 0 0-.734.568c-2.58 8.948 1.538 20.807 14.226 25.163l-.024-.048Z"
              fill="#642DB3"
            />
          </svg>
        </div>
        <div
          class="Spacing__SpaceTop-sc-11df7di-1 fckOgK"
        />
        <div
          aria-level="3"
          class="sc-kAyceB kJJjGB"
          role="heading"
        >
          No encontramos resultados
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
